ops/terraform/services/migrator/Jenkinsfile

```diff
@@ -0,0 +1,242 @@
+        return false
+    }
+
+    // plan/apply terraform
+    executeTerraform(amiId: amiId,
+                     bfdEnv: bfdEnv,
+                     gitBranchName: gitBranchName,
+                     heartbeatInterval: heartbeatInterval,
+                     createMigratorInstance: true)
+
+    // monitor migrator deployment
+    finalMigratorStatus = monitorMigrator(
+        sqsQueueName: sqsQueueName,
+        region: awsRegion,
+        heartbeatInterval: heartbeatInterval,
+        maxMessages: 10,
+        authFunction
+    )
+
+    // re-authenticate
+    authFunction()
+
+    // set return value for final disposition
+    if (finalMigratorStatus == '0') {
+        migratorDeployedSuccessfully = true
+        // Teardown when there is a healthy exit status
+        executeTerraform(amiId: amiId, bfdEnv: bfdEnv, gitBranchName: gitBranchName)
+    } else {
+        migratorDeployedSuccessfully = false
+    }
+
+    purgeQueue(sqsQueueName)
+
+    println "Migrator completed with exit status ${finalMigratorStatus}"
+    return migratorDeployedSuccessfully
+}
+
+// wraps the aws cli's receive-message subcommand to produce JSON objects
+// that represent the deployed migrator's state. Contains the following
+// fields mapped to string values: pid, start_time, stop_time, status, code
+def receiveMigratorMessages(Map args = [:]) {
+    maxMessages = args.maxMessages
+    region = args.region
+    url = args.url
+    visibilityTimeoutSeconds = args.visibilityTimeoutSeconds
+    waitTimeSeconds = args.waitTimeSeconds
+
+    withEnv(["url=${url}",
+             "region=${region}",
+             "maxMessages=${maxMessages}",
+             "waitTimeSeconds=${waitTimeSeconds}",
+             "visibilityTimeoutSeconds=${visibilityTimeoutSeconds}"]) {
+        rawMessages = sh(
+            returnStdout: true,
+            script: '''
+#!/usr/bin/env bash
+aws sqs receive-message \
+  --region "$region" \
+  --queue-url "$url" \
+  --max-number-of-messages "$maxMessages" \
+  --wait-time-seconds "$waitTimeSeconds" \
+  --visibility-timeout "$visibilityTimeoutSeconds" \
+  --output json \
+  --query Messages |\
+jq '.? | map({receipt: .ReceiptHandle, body: .Body | fromjson}) | unique'
+'''
+        ).trim()
+    }
+    try {
+        jsonMessages = readJSON text: rawMessages
+    } catch(err) {
+        jsonMessages = readJSON text: '[]'
+    }
+    return jsonMessages
+}
+
+// deletes a message by the given `receipt` which indicates the associated message has been proceesed
+String deleteMessage(String receipt, String url, String region = 'us-east-1') {
+    result = sh(returnStdout: true, script: "aws sqs delete-message --queue-url ${url} --region ${region} --receipt-handle ${receipt}")
+    return result
+}
+
+// returns true when if the given `sqsQueueName` exists
+boolean sqsQueueExists(String sqsQueueName, String region = 'us-east-1') {
+    // For whatever reason, the ternary expression doesn't work here. I still don't understand Groovy.
```

Comment:
what was the error?

Reply:
I don't recall, but doing something like the below just didn't behave as I expected. I've gone back and forth in my process and it might have been something entirely unrelated to the ternary expression. Saving a few lines might be nice, though.
```groovy
    exists = sh(returnStatus: true, script: "aws sqs get-queue-url --queue-name ${sqsQueueName} --region ${awsRegion} --output text")
    return exists == '0' : true ? false
```